collector/ethtool_linux.go

```diff
@@ -116,6 +118,25 @@ func init() {
 	registerCollector("ethtool", defaultDisabled, NewEthtoolCollector)
 }
 
+// Sanitize the given metric name.
+// OpenMetrics and the Prometheus exposition format require the metric name
+// to consist only of alphanumericals and "_", ":" and they must not start
+// with digits. If the metric name starts with a digit, add "_" as prefix.
+func SanitizeMetricName(metricName string) string {
+	sanitized := metricNameRegex.ReplaceAllString(metricName, "_")
+	if len(sanitized) >= 1 && sanitized[0] >= '0' && sanitized[0] <= '9' {
+		sanitized = "_" + sanitized
+	}
+	return sanitized
+}
+
+// Generate the fully-qualified metric name for the ethool metric.
+func buildEthtoolFQName(metric string) string {
+	metricFQName := strings.ToLower(SanitizeMetricName(prometheus.BuildFQName(namespace, "ethtool", metric)))
```

Comment:
BuildFQName should not inherently need to be sanitized if the namespace and subsystem are legal strings. I think you only need to sanitize the `metric` variable here, since that is the unknown that comes from the driver. It would also mean that you could drop the first char sanity check.

Reply:
Okay. I changed the code to only sanitize `metric`. This requires the `tx` and `rx` replacements to be more clever and leading underscores need to be stripped.